### PR TITLE
Improve MySQL detection

### PR DESCRIPTION
**Pull request by 0xA50C1A1**
Please sign (check) the below before submitting the Pull Request:

- [x] I have signed the ntop Contributor License Agreement at https://github.com/ntop/legal/blob/main/individual-contributor-licence-agreement.md
- [x] I have read the contributing guide lines at https://github.com/ntop/nDPI/blob/dev/CONTRIBUTING.md
- [x] I have updated the documentation (in doc/) to reflect the changes made (if applicable)

Describe changes:

Slight improvement of MySQL dissector: added MariaDB support and more stringent checks.

### CODE REVIEW

File: src/lib/protocols/mysql.c [hunk 1]
```diff
@@ -30,50 +31,27 @@
...
+    if ((u_int32_t)(packet->payload_packet_len-4) == length && 
+        packet->payload[4] == 0x0A && ((memcmp(&packet->payload[5], "5.5.5-", 6) == 0) || 
+        (packet->payload[5] > 0x33 && packet->payload[5] < 0x39)))
+    {
+      if ((memcmp(&packet->payload[packet->payload_packet_len-10], "_password", 9) == 0) ||
+          (memcmp(&packet->payload[packet->payload_packet_len-10], "_kerberos", 9) == 0) ||
+          (memcmp(&packet->payload[packet->payload_packet_len-9], "_windows", 8) == 0) ||
+          (memcmp(&packet->payload[packet->payload_packet_len-8], "_simple", 7) == 0) ||
+          (memcmp(&packet->payload[packet->payload_packet_len-8], "_gssapi", 7) == 0) ||
+          (memcmp(&packet->payload[packet->payload_packet_len-5], "_pam", 4) == 0))
```
Comment (0xA50C1A1) [inline src/lib/protocols/mysql.c:50]: Is it worth to add checks for other [authentication plugins](https://dev.mysql.com/doc/refman/8.0/en/authentication-plugins.html)?

**Comment by 0xA50C1A1** [comment]
@IvanNardi @utoni guys, what do you think about this? I bumped the minimum MySQL version in check from 1 to 4 since there's not many people using these ancient relics.

Well, also authentication plugin support was introduced only in MySQL 4, if memory serves me right.

File: src/lib/protocols/mysql.c (hunk 1 above)
Comment (IvanNardi) [reply src/lib/protocols/mysql.c:50]: I think that it is fine as-is, unless you have some traces with the other plugins

**Review by IvanNardi — APPROVED** [review]
Thanks

**Pull request merged**